### PR TITLE
Fix from_address in L1-L2 messaging

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- Combines #296 and #298 
  - The target branch of this PR is the target branch of those PRs; when they are merged, this one can be merged, too.
- Fixes some situations of messaging when the message is sent by a contract call, and not a library call
  - Relies on storage address to achieve this.

## Development related changes

- When using the script `run_e2e.sh`, it's important to use scarb `2.3.1` for now as starkli is not release with `2.4.0` compiler version.
- Improved `run_e2e.sh` and the corresponding README.md
- Refactors flush assertion to assert body content with a single `assert_eq!`

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: contracts/l1-l2-messaging/run_e2e.sh [hunk 1]
```diff
@@ -20,28 +23,30 @@ pushd ./solidity
 forge script ./script/L1L2.s.sol:Deploy --broadcast --rpc-url $ETH_RPC_URL
 popd
 
+MAX_FEE=99999999999999999999
+
 # Declare and deploy l1-l2 contract on L2.
-starkli declare ./cairo/target/dev/cairo_l1_l2.contract_class.json
-starkli deploy 0x0211fd0483be230ba40d43f51bd18ae239b913f529f95ce10253e514175efb3e --salt 123
+starkli declare ./cairo/target/dev/cairo_l1_l2.contract_class.json --max-fee-raw $MAX_FEE
+starkli deploy 0x0211fd0483be230ba40d43f51bd18ae239b913f529f95ce10253e514175efb3e --salt 123 --max-fee-raw $MAX_FEE
```
Comment (ivpavici) [inline contracts/l1-l2-messaging/run_e2e.sh:30]: should we extract also `0x0211fd0483be230ba40d43f51bd18ae239b913f529f95ce10253e514175efb3e` to a constant together with others?

File: contracts/l1-l2-messaging/run_e2e.sh [hunk 2]
```diff
@@ -60,13 +65,14 @@ curl -H 'Content-Type: application/json' \
...
 # Send back some balance to consume manually.
-starkli invoke 0x03c80468c8fe2fd36fadf1b484136b4cd8a372f789e8aebcc6671e00101290a4 withdraw 0x1 0x2 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
+echo "INVOKE"
+starkli invoke $CONTRACT_L2 withdraw 0x1 0x2 $CONTRACT_L1 --max-fee-raw $MAX_FEE
 
 curl -H 'Content-Type: application/json' \
      -d '{
-          "from_address": "0x34ba56f92265f0868c57d3fe72ecab144fc96f97954bbbc4252cef8e8a979ba",
-          "to_address": "0xe7f1725e7734ce288f8367e1bb143e90bb3f0512", "payload": ["0x0","0x1","0x2"]
+          "from_address": "0x03c80468c8fe2fd36fadf1b484136b4cd8a372f789e8aebcc6671e00101290a4",
```
Comment (ivpavici) [inline contracts/l1-l2-messaging/run_e2e.sh:73]: can the constants be embedded here as well instead?

Comment (FabijanC) [reply contracts/l1-l2-messaging/run_e2e.sh]: Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/299/commits/1c332514eb30341dde6eb9fe743d07450c6ba77c

File: contracts/l1-l2-messaging/run_e2e.sh (hunk 1 above)
Comment (FabijanC) [reply contracts/l1-l2-messaging/run_e2e.sh]: Added comment in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/299/commits/22f4ec9be263daabb71c6dc8cad998b2e0b85c7c

**Review by ivpavici — APPROVED** [review]

**Pull request merged**